components/server/src/server.rs

```diff
@@ -341,24 +341,28 @@ where
 
         let resource_manager = if config.resource_control.enabled {
             let mgr = Arc::new(ResourceGroupManager::default());
-            let mut resource_mgr_service =
-                ResourceManagerService::new(mgr.clone(), pd_client.clone());
+            let resource_mgr_service = ResourceManagerService::new(mgr.clone(), pd_client.clone());
```

Comment:
How about extracting the background tasks spawning to resource group manager, so we don't need to write it twice

Reply:
done.